pygmt/datasets/load_earth_dataset.py

```diff
@@ -0,0 +1,181 @@
+"""
+Internal function to download remote Earth datasets.
+"""
+from typing import Dict, NamedTuple
+
+from pygmt.exceptions import GMTInvalidInput
+from pygmt.helpers import kwargs_to_strings
+from pygmt.io import load_dataarray
+from pygmt.src import grdcut, which
+
+
+class Resolution(NamedTuple):
+    """
+    The available grid registrations for a given resolution and whether it is a
+    tiled grid.
+    """
+
+    registrations: list[str]
+    tiled: bool
+
+
+class Dataset(NamedTuple):
+    """
+    Standard information about a dataset and grid metadata.
+    """
+
+    dataset_title: str
+    name: str
+    long_name: str
+    units: str
+    resolutions: Dict[str, Resolution]
+    vertical_datum: str
+    horizontal_datum: str
+
+
+earth_relief_resolutions = {
+    "01d": Resolution(["pixel", "gridline"], False),
+    "30m": Resolution(["pixel", "gridline"], False),
+    "20m": Resolution(["pixel", "gridline"], False),
+    "15m": Resolution(["pixel", "gridline"], False),
+    "10m": Resolution(["pixel", "gridline"], False),
+    "06m": Resolution(["pixel", "gridline"], False),
+    "05m": Resolution(["pixel", "gridline"], True),
+    "04m": Resolution(["pixel", "gridline"], True),
+    "03m": Resolution(["pixel", "gridline"], True),
+    "02m": Resolution(["pixel", "gridline"], True),
+    "01m": Resolution(["pixel", "gridline"], True),
+    "30s": Resolution(["pixel", "gridline"], True),
+    "15s": Resolution(["pixel"], True),
+    "03s": Resolution(["gridline"], True),
+    "01s": Resolution(["gridline"], True),
+}
+
+earth_age_resolutions = {
+    "01d": Resolution(["pixel", "gridline"], False),
+    "30m": Resolution(["pixel", "gridline"], False),
+    "20m": Resolution(["pixel", "gridline"], False),
+    "15m": Resolution(["pixel", "gridline"], False),
+    "10m": Resolution(["pixel", "gridline"], False),
+    "06m": Resolution(["pixel", "gridline"], False),
+    "05m": Resolution(["pixel", "gridline"], True),
+    "04m": Resolution(["pixel", "gridline"], True),
+    "03m": Resolution(["pixel", "gridline"], True),
+    "02m": Resolution(["pixel", "gridline"], True),
+    "01m": Resolution(["gridline"], True),
+}
+
+
+datasets = {
+    "earth_relief": Dataset(
+        dataset_title="Earth relief",
+        name="elevation",
+        long_name="elevation relative to the geoid",
+        units="meters",
+        vertical_datum="EMG96",
+        horizontal_datum="WGS84",
+        resolutions=earth_relief_resolutions,
```

Comment:
I think having all metadata defined in a single place can make it easier to maintain.
```suggestion
        resolutions={
            "01d": Resolution(["pixel", "gridline"], False),
            "30m": Resolution(["pixel", "gridline"], False),
            "20m": Resolution(["pixel", "gridline"], False),
            "15m": Resolution(["pixel", "gridline"], False),
            "10m": Resolution(["pixel", "gridline"], False),
            "06m": Resolution(["pixel", "gridline"], False),
            "05m": Resolution(["pixel", "gridline"], True),
            "04m": Resolution(["pixel", "gridline"], True),
            "03m": Resolution(["pixel", "gridline"], True),
            "02m": Resolution(["pixel", "gridline"], True),
            "01m": Resolution(["pixel", "gridline"], True),
            "30s": Resolution(["pixel", "gridline"], True),
            "15s": Resolution(["pixel"], True),
            "03s": Resolution(["gridline"], True),
            "01s": Resolution(["gridline"], True),
        },
```